compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -472,11 +469,19 @@ object Contexts {
       this
     }
 
+    def reuseIn(outer: Context): this.type =
+      implicitsCache = null
+      phasedCtxs = null
+      sourceCtx = null
```

Comment:
This could be done in `init`, that way we only have one correct way to initialize things, no matter if it's the first or a latter initialization.

Reply:
I did it here in order to avoid the additional stores in `init`. We already have
```
implicitsCache = null
```
in the definition. I assume that the JVM will be smart enough to avoid the store since the object is pre-initialzied to zeroes anyway. But i don't think it can avoid the store in a separate initializer method.